### PR TITLE
Feature #52: Implement mapping of OSM charging values

### PR DESCRIPTION
**Pull request by cs-dieter-kling**

### CODE REVIEW

File: charging_stations_pipelines/pipelines/osm/osm_receiver.py [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+
+    :param country_code: The country code of the desired country.
+    :type country_code: str
+    :param tmp_data_path: The path to save the downloaded OSM data.
+    :type tmp_data_path: str
+    :return: None
+    """
     country_code_to_area = {
         "DE": "Deutschland",
         "FR": "France métropolitaine",
```
Comment (stefanpernpaintner) [inline charging_stations_pipelines/pipelines/osm/osm_receiver.py:34]: Do we miss other countries like Italy, GB, NOR, SWE ?

File: charging_stations_pipelines/pipelines/osm/osm_receiver.py [hunk 2]
```diff
@@ -45,4 +70,4 @@ def get_osm_data(country_code: str, tmp_data_path):
 
 
 if __name__ == "__main__":
-    get_osm_data(tmp_data_path="./osm_france.json")
+    get_osm_data("FR", "./osm_france.json")
```
Comment (stefanpernpaintner) [inline charging_stations_pipelines/pipelines/osm/osm_receiver.py:73]: Is this main function needed? I think we can remove it.

Comment (stgrmks) [reply charging_stations_pipelines/pipelines/osm/osm_receiver.py:73]: Typically used for local testing. Should be removed.

Comment (d3v3l0) [reply charging_stations_pipelines/pipelines/osm/osm_receiver.py]: OK, I removed call as main.

File: charging_stations_pipelines/pipelines/osm/osm_receiver.py (hunk 1 above)
Comment (d3v3l0) [reply charging_stations_pipelines/pipelines/osm/osm_receiver.py:36]: No, they are all there, the list is just not shown in full length.

File: test/pipelines/osm/test_osm_mapper.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                "amperage": "125 A - DC",
+            },
+        }
+        charging: Charging = map_charging_osm(json_data, 1)
+        self.assertEqual([125.0], charging.ampere_list)
+
+    def test_charging_mapping__amperage_04(self):
+        json_data = {
+            "tags": {
+`                "amperage": "16;32",
```
Comment (stefanpernpaintner) [inline test/pipelines/osm/test_osm_mapper.py:110]: The backticks in line 110 seem to be a typo.

**Comment by stefanpernpaintner** [comment]
When executing the tests I run in the following errors: 
```
======================================================================
ERROR: test_address_mapping (pipelines.osm.test_osm_mapper.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pernpas/encrypted/dev/projects/stations-pipelines/test/pipelines/osm/test_osm_mapper.py", line 38, in test_address_mapping
    address: Address = map_address_osm(json_data, 1)
  File "/home/pernpas/encrypted/dev/projects/stations-pipelines/charging_stations_pipelines/pipelines/osm/osm_mapper.py", line 73, in map_address_osm
    map_address.district_old = None
  File "/home/pernpas/encrypted/dev/projects/stations-pipelines/charging_stations_pipelines/models/__init__.py", line 21, in __setattr__
    raise AttributeError(f"Cannot set non-existing attribute '{name}' on class '{self.__class__.__name__}'.")
AttributeError: Cannot set non-existing attribute 'district_old' on class 'Address'.

======================================================================
FAIL: test_charging_mapping__plain_01 (pipelines.osm.test_osm_mapper.Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pernpas/encrypted/dev/projects/stations-pipelines/test/pipelines/osm/test_osm_mapper.py", line 72, in test_charging_mapping__plain_01
    self.assertEqual([2.3], charging.kw_list)
AssertionError: Lists differ: [2.3] != []

First list contains 1 additional elements.
First extra element 0:
2.3

- [2.3]
+ []

```
Can you please check?

File: charging_stations_pipelines/pipelines/osm/osm_mapper.py [hunk 4]
```diff
@@ -1,23 +1,34 @@
 import logging
+import math
+import math
```
Comment (stefanpernpaintner) [inline charging_stations_pipelines/pipelines/osm/osm_mapper.py:3]: duplicated import of math

File: charging_stations_pipelines/pipelines/osm/osm_mapper.py [hunk 5]
```diff
@@ -28,35 +39,176 @@ def map_station_osm(entry: Dict, country_code: str):
...
+        street = tags["addr:street"]
+
+        map_address = Address()
+        # TODO add state_old to Address
+        # map_address.state_old = None
+        map_address.station_id = station_id
+        map_address.street = street + " " + housenumber
+        map_address.town = city
+        # TODO add postcode to Address
+        # map_address.postcode = postcode
```
Comment (stefanpernpaintner) [inline charging_stations_pipelines/pipelines/osm/osm_mapper.py:72]: Why is this commented? I assume there is an issue with this field.

File: test/pipelines/osm/test_osm_mapper.py (hunk 3 above)
Comment (cs-dieter-kling) [reply test/pipelines/osm/test_osm_mapper.py]: Fixed.

File: charging_stations_pipelines/pipelines/osm/osm_mapper.py (hunk 5 above)
Comment (cs-dieter-kling) [reply charging_stations_pipelines/pipelines/osm/osm_mapper.py]: I clarified this with @mjmader... this field should be used (again).

**Comment by cs-dieter-kling** [comment]
Hi @stefanpernpaintner, all problems are fixed now.

File: charging_stations_pipelines/pipelines/osm/osm_mapper.py [hunk 6]
```diff
@@ -1,23 +1,45 @@
...
 
 logger = logging.getLogger(__name__)
 
+SOCKET_TYPES = {  # type: Final[dict[str, str]]
+    'socket:chademo':            'CHAdeMO',
+    'socket:schuko':             'AC Schuko',
+    'socket:tesla_supercharger': 'Tesla Supercharger',
+    'socket:type2':              'AC Steckdose Typ 2',
+    'socket:type2:combo':        'AC Steckdose Typ 2 Combo',
+    'socket:type2_combo':        'AC Steckdose Typ 2 Combo',
```
Comment (stefanpernpaintner) [inline charging_stations_pipelines/pipelines/osm/osm_mapper.py:25]: Socket Type2 Combo is not an AC Steckdose. From my point of view it has to be mapped to CCS Typ 2 which is capable of AC and DC

Comment (stefanpernpaintner) [reply charging_stations_pipelines/pipelines/osm/osm_mapper.py:25]: image

Comment (cs-dieter-kling) [reply charging_stations_pipelines/pipelines/osm/osm_mapper.py]: Fixed.

**Pull request merged**